### PR TITLE
Cherry pick changes from master into next

### PR DESCRIPTION
**Pull request by JustinBeckwith**

### CODE REVIEW

File: src/auth/oauth2client.ts [hunk 1]
```diff
@@ -17,18 +17,20 @@
...
 import * as http from 'http';
 import * as querystring from 'querystring';
-
 import {PemVerifier} from './../pemverifier';
 import {BodyResponseCallback} from './../transporters';
 import {AuthClient} from './authclient';
 import {CredentialRequest, Credentials} from './credentials';
 import {LoginTicket} from './loginticket';
 
+const isString = require('lodash.isstring');
```
Comment (gaofanmichael) [inline src/auth/oauth2client.ts]: Can we use import signature on this? `@sindresorhus/is` can be an alternative as well.

Comment (ofrobots) [reply src/auth/oauth2client.ts]: Let's open this as a bug. A cherry-pick between branches shouldn't edit.

**Comment by ofrobots** [comment]
@JustinBeckwith an alternative would be to rebase the next branch onto master. WDYT? If that sounds reasonable, go ahead and do that, no need to PR.

File: src/auth/oauth2client.ts (hunk 1 above)
Comment (JustinBeckwith) [reply src/auth/oauth2client.ts]: FWIW I did try to do that :) Couldn't get it to work and decided to come back later.

**Comment by ofrobots** [comment]
I have a rebased `n2` branch on my fork. I can replace the `next` branch once the pending PRs land. Closing this PR.

**Pull request closed without merging**